drivers/SmartThings/matter-sensor/src/init.lua

```diff
@@ -166,11 +228,60 @@ local function humidity_attr_handler(driver, device, ib, response)
   end
 end
 
+local BOOLEAN_CAP_EVENT_MAP = {
+  [true] = {
+      ["WATER_FREEZE_DETECTOR"] = capabilities.temperatureAlarm.temperatureAlarm.freeze(),
+      ["WATER_LEAK_DETECTOR"] = capabilities.waterSensor.water.wet(),
+      ["RAIN_SENSOR"] = capabilities.rainSensor.rain.detected(),
+      ["CONTACT_SENSOR"] =  capabilities.contactSensor.contact.closed(),
+  },
+  [false] = {
+      ["WATER_FREEZE_DETECTOR"] = capabilities.temperatureAlarm.temperatureAlarm.cleared(),
+      ["WATER_LEAK_DETECTOR"] = capabilities.waterSensor.water.dry(),
+      ["RAIN_SENSOR"] = capabilities.rainSensor.rain.undetected(),
+      ["CONTACT_SENSOR"] =  capabilities.contactSensor.contact.open(),
+  }
+}
+
 local function boolean_attr_handler(driver, device, ib, response)
+  local name = nil
+  for dt_name, _ in pairs(BOOLEAN_DEVICE_TYPE_INFO) do
+      local dt_ep_id = device:get_field(dt_name)
+      if ib.endpoint_id == dt_ep_id then
+          name = dt_name
+          break
+      end
+  end
+  if name == nil then
+    -- The generic case where no device type has been specified but the profile uses this capability.
+    if device:supports_capability(capabilities.contactSensor) then
+      device:emit_event_for_endpoint(ib.endpoint_id, BOOLEAN_CAP_EVENT_MAP[ib.data.value]["CONTACT_SENSOR"])
+    else
+      log.error("No Boolean device type found on an endpoint, BooleanState handler aborted")
+    end
+    return
+  end
+  device:emit_event_for_endpoint(ib.endpoint_id, BOOLEAN_CAP_EVENT_MAP[ib.data.value][name])
+end
+
+local function supported_sensitivities_handler(driver, device, ib, response)
   if ib.data.value then
-    device:emit_event_for_endpoint(ib.endpoint_id, capabilities.contactSensor.contact.closed())
+    device:set_field(MAX_SENSITIVITY_LEVEL, ib.data.value)
+    device:set_field(MIN_SENSITIVITY_LEVEL, 0x00)
+  end
+end
+
+local function default_sensitivity_handler(driver, device, ib, response)
```

Comment:
I don't think we should write the default sensitivity level to the device. Just because it is the default doesn't mean that's what it's current value should be. For example, what if the value was already changed from the default before the device is onboarded?

You can probably just remove this handler all together

Reply:
Good point